hack/lib/golang.sh

```diff
@@ -146,9 +145,11 @@ readonly KUBE_TEST_PORTABLE=(
   hack/lib
 )
 
-# Node test has built-in etcd and kube-apiserver, it can only be built on the
-# same platforms with kube-apiserver.
+# Test targets which run on the Kubernetes clusters directly, so we only
+# need to target server platforms.
+# These binaries will be distributed in the kubernetes-test tarball.
 readonly KUBE_NODE_TEST_TARGETS=(
+  cmd/kubemark
```

Comment:
Unless I'm really misunderstanding this, `cmd/kubemark` is the hollow-node binary, which gets built into the kubemark container image (cluster/images/kubemark) which is then run on the kubernetes cluster under test, hence we should target platforms that run kubernetes.

I could probably rename this target to `KUBE_SERVER_TEST_TARGETS` if it'd be a little clearer. That'd at least match the other names here (KUBE_SERVER and KUBE_CLIENT).